### DISCUSSION TITLE
Issue with print clearing sequence being its own "print"

### DISCUSSION THREAD

**Discussion by whopster17** [smartin015/continuousprint#111] [Q&A] [answered]
Basically, on the older version of Continuous Print, I would have an IFTTT plugin that would send a webhook that would trigger a smart plug to turn on a fan to cool down the print bed. After 15 minutes, the fan would turn off and the bed clearing would start.

Now, on the newer versions, the bed clearing script is considered its own "print". Because of this, immediately after the bed is cleared, the fan turns on because the "gcode" finished. So for 15 minutes, the fan is on for the beginning of the print which is definitely not ideal. I tried to look for older releases but could only find so far back. Is there any other way I could workaround this?

**Comment by smartin015** [comment, marked as answer]
Is the plugin you're using [this OctoPrint-IFTTT plugin](https://plugins.octoprint.org/plugins/IFTTT/)? It looks like it just [does a string match](https://github.com/tjjfvi/OctoPrint-IFTTT/blob/master/octoprint_IFTTT/__init__.py#L23) on events. 

I can't override the way OctoPrint fires print completion events - that's built in. However, I could add a [custom event](https://docs.octoprint.org/en/master/plugins/hooks.html#sec-plugins-hook-events-register-custom-events) which fires when CPQ starts the bed clearing process. In that case, all you'd have to do is change the event name in the IFTTT plugin to match the custom event, and ensure your bed clearing script appropriately waits for the proper time/temperature.

Let me know if this works for you, and I'll open a feature request.

**Reply by whopster17** [reply]
This would be absolutely perfect my friend and yes that's the IFTTT plugin I'm using. This would be so useful!

**Reply by smartin015** [reply]
Great - I opened #117, please keep an eye out there for implementation progress.
